DNN Platform/DotNetNuke.Abstractions/Logging/ILogTypeConfigInfo.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+namespace DotNetNuke.Abstractions.Logging
+{
+    using System;
+
+    /// <summary>
+    /// Log type configuration info.
+    /// </summary>
+    public interface ILogTypeConfigInfo : ILogTypeInfo
+    {
+        /// <summary>
+        /// Gets the start <see cref="DateTime"/>.
+        /// </summary>
+        DateTime StartDateTime { get; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether an email notifiaction is active.
+        /// </summary>
+        bool EmailNotificationIsActive { get; set; }
+
+        /// <summary>
+        /// Gets or sets the email from address.
+        /// </summary>
+        string MailFromAddress { get; set; }
+
+        /// <summary>
+        /// Gets or sets the email to address.
+        /// </summary>
+        string MailToAddress { get; set; }
+
+        /// <summary>
+        /// Gets or sets the notification threshold.
+        /// </summary>
+        int NotificationThreshold { get; set; }
+
+        /// <summary>
+        /// Gets or sets the notification threshold time.
+        /// </summary>
+        int NotificationThresholdTime { get; set; }
+
+        /// <summary>
+        /// Gets or sets the notification threshold time.
+        /// </summary>
+        NotificationThresholdTimeType NotificationThresholdTimeType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Id.
+        /// </summary>
+        string Id { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether if logging is active.
+        /// </summary>
+        bool LoggingIsActive { get; set; }
+
+        /// <summary>
+        /// Gets or sets the log file name.
+        /// </summary>
+        string LogFileName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the log file name with path.
+        /// </summary>
+        string LogFileNameWithPath { get; set; }
+
+        /// <summary>
+        /// Gets or sets the log type portal Id.
+        /// </summary>
+        string LogTypePortalId { get; set; }
```

Comment:
@mitchelsellers @bdukes Do you know why this property is a `string`? Since the `PortalId` is type of `int` should we update this for type safety?

Otherwise code might look like this
```c#
ILogTypeConfigInfo configInfo = new LogTypeConfigInfo();
configInfo.LogTypePortalId = $"{PortalSettings.PortalID}";
```

Reply:
To accomplish this we would need to add a new property `AllowOnAllPortals` something to indicate the wildcard.

I retract my statement and don't think we should make any changes. I'll leave this conversation open for a little bit in case anyone wants to comment on it

Reply:
Adding a second property could easily lead into inconsistency. I suggest using a dedicated value instead - either Integer.Null or -1 as usual.

Reply:
Thanks @sleupold for the input but I think it is best to just leave this as is for now. I don't want to make SQL changes to the database and don't think we should be making any changes to stuff like this unless we are doing a major overhaul of the database

Reply:
@ahoefling I agree, but IMO object properties should be mapped 1:1 to database columns (or rows, if stored serialized, lke settings), otherwise it becomes confusing or even inconsistent. 
If we don't modify the database, AllowOnAllPortals wuld need to be added as a calculated column based on PortalID.

Reply:
@sleupold we are in agreement, technically the `string LogTypePortalId` is mapped 1:1 to the database column. It lacks any foreign key constraint and can't be indexed correctly.  This is a classic example of a SQL anti-pattern, but it has been in DNN for so long it will cause more harm today to fix it and make a breaking change.